### ISSUE TITLE
[feature] Load images from Unsplash

### ISSUE THREAD

**Issue by joaomlneto** [matthewlucock/pure-atom#4]
Currently we can get background images from *Default images* (which are bundled/hardcoded) and *Custom images* (which are user-defined).

My suggestion for improvement is to add [Unsplash](https://unsplash.com) as a third option.

They offer [*Unsplash Source*](https://source.unsplash.com), which is a minimalistic API to get photos.

I may attempt do it when I have the time; but for now I'm not offering myself ;-)

###### Hopefully this is not considered an ad! :-)

Also, many thanks for this theme! I love it.

**Comment by matthewlucock** [comment]
I'm glad somebody else likes it besides me! 😛 This API is actually pretty cool; I had no idea about it.

How exactly do you envision this working? Would it just load a random image, or would the image selection view provide some kind of interface for accessing the API, or something else? What's the user experience here?

**Comment by joaomlneto** [comment]
> Would it just load a random image, or would the image selection view provide some kind of interface for accessing the API, or something else?

Default behavior vs customizability? Ideally, both :-) we could have a v1 that just loads a random image.

> How exactly do you envision this working?

> What's the user experience here?

My `--verbose` vision:

1. Add `unsplash` option in the *Select new images from* dropdown. Maybe make it a checklist so that the user can select any combination of options.

Default behavior could be either:
- Load random image (`https://source.unsplash.com/random`)
- Load image from a predefined category/*search term* of our choosing (e.g. `nature`)
- Load image from a predefined user (e.g. you could create an account and curate some default set of images for this theme; similar to what you do already with the `Default images`... but then it is a bit weird if we get the same set of images from two different sources).

2. Configuration:
- `v1`: Have a textbox in the settings (if empty then use default behavior; otherwise consider it a *search term* (e.g. `nature`)).
- `v2`: Dropdown box with a few predefined options (`default`, `nature`, `ocean`, …) as well as `custom user` and `custom terms`. If either of the latter two are selected, have an additional textbox for the user to write the userID or parameters. But this is overkill :P I'd be 120% happy with v1.

3. Behind the scenes:
- Automatically detect the resolution, so we can pass it to the API.
- Fallback to *predefined images* if we can't get images from Unsplash; or simply reuse the ones we already downloaded.
- Cache a few images so that we don't immediately fallback to *predefined images* if we're without internet

**Comment by matthewlucock** [comment]
Sorry for taking so long to get back to this. I’ve started working full-time hours as a web developer and so I’ve suddenly become pretty busy.

Maybe you’ve had a different experience to me adding custom images to the theme, but, as I reference in the README, I am strongly of the view that the background images must be curated, otherwise it is fairly likely that they just won’t look good in Pure. The majority of images, in my experience, are unsuitable, and that suitability is not determined by the genre of the image (e.g nature), but by particular visual characteristics of the image. I am unconvinced of the benefit that loading random images from an external source would provide. I can understand why there might be a perceived benefit given that the fixed set of default images could become stale over time, but this is one of the big reasons why I implemented the ability to add custom images. Even if a user curates a set of images, I don’t really see why it would be advantageous to load those images from an external source as opposed to just adding them as custom images to Pure.

**Comment by joaomlneto** [comment]
> Sorry for taking so long to get back to this

Not at all an issue! :-)

------

**TL;DR: I agree with you -- it's quite dubious that this is a good idea; and even if it is, it's probably not much more versatile than allowing for user-defined custom images.**

> The majority of images, in my experience, are unsuitable, and that suitability is not determined by the genre of the image (e.g nature), but by particular visual characteristics of the image.

The set of images that I've checked in Unsplash seems to be quite good. But given the randomness it's quite probably that a few images will be "bad".

> I am strongly of the view that the background images must be curated

An alternative would be to set up a source (not necessarily in Unsplash) that is curated by you. An external source allows for a large set of images that can be updated over time.

> Even if a user curates a set of images, I don’t really see why it would be advantageous to load those images from an external source as opposed to just adding them as custom images to Pure.

The only upside is if people get bored of the images **and** are lazy to curate/add their own :-) Personally I'll avoid doing it if I can have an unlimited source of great images as the ones included here.

###### Sorry; I wrote this a bit in a hurry!
